.github/workflows/test-sqlite.yml

```
@@ -0,0 +1,24 @@
+name: test-sqlite
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
```

```suggestion
permissions:
  contents: read

```

though it is possible that is the default now.

---

I think it is. Going to skip this on the principle that fewer lines is better.

---

It looks like the default is `read-all`, which is more general (e.g., it allows access to readable properties of the repo besides the files), but that is probably not a big issue.